var/ramble/repos/builtin/modifiers/intel-aps/modifier.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2022-2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from ramble.modkit import *  # noqa: F403
+
+
+class IntelAps(SpackModifier):
+    """Define a modifier for Intel's Application Performance Snapshot
+
+    Intel's Application Performance Snapshot (APS) is a high level profiler. It
+    gives a quick view into the high level performance characteristics of an
+    experiment. This modifier allows for easy application of APS to experiments.
+    """
+    name = "intel-aps"
+
+    tags = ['profiler', 'performance-analysis']
+
+    mode('mpi', description='Mode for collecting mpi statistics')
+
+    variable_modification('aps_log_dir', 'aps_{executable_name}_results_dir',
+                          method='set', modes=['mpi'])
+    variable_modification('aps_flags', '-c mpi -r {aps_log_dir}', method='set', modes=['mpi'])
+    variable_modification('mpi_command', 'aps {aps_flags}', method='prepend', modes=['mpi'])
+
+    archive_pattern('aps_*_results_dir/*')
+
+    software_spec('intel-oneapi-vtune', spack_spec='intel-oneapi-vtune')
+
+    required_package('intel-oneapi-vtune')
+
+    executable_modifier('aps_summary')
+
+    def aps_summary(self, executable_name, executable):
+        from ramble.util.executable import CommandExecutable
+
+        pre_exec = []
+        post_exec = []
+        if executable.mpi:
+            post_exec.append(
+                CommandExecutable(f'gen-aps-{executable_name}',
+                                  template=['echo "APS Results for executable {executable_name}"',
+                                            'aps-report -s -D {aps_log_dir}'],
+                                  mpi=False,
+                                  redirect='{log_file}'
+                                  )
+            )
+
+        return pre_exec, post_exec
+
+    figure_of_merit_context('APS Executable',
```

Comment:
We talked about this offline a bit, but I'm not in love with a few things around how the FOMS are displayed. 

I think using a context to nest/mark these as "coming from APS" is a neat trick, but it feels a bit like an abuse of contexts and not really their intended purpose 

Overall I'd do wonder if FOMs from mods should be more clearly marked, something like `app::mod::fom_name` to give a more structural nesting